src/test/abilities/illusion.test.ts

```diff
@@ -0,0 +1,127 @@
+import { Abilities } from "#enums/abilities";
+
+describe("Abilities - Illusion", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    vi.spyOn(overrides, "BATTLE_TYPE_OVERRIDE", "get").mockReturnValue("single");
+    vi.spyOn(overrides, "OPP_SPECIES_OVERRIDE", "get").mockReturnValue(Species.ZORUA);
+    vi.spyOn(overrides, "OPP_ABILITY_OVERRIDE", "get").mockReturnValue(Abilities.ILLUSION);
+    vi.spyOn(overrides, "OPP_MOVESET_OVERRIDE", "get").mockReturnValue([Moves.TACKLE, Moves.TACKLE, Moves.TACKLE, Moves.TACKLE]);
+    vi.spyOn(overrides, "OPP_HELD_ITEMS_OVERRIDE", "get").mockReturnValue([{name: "WIDE_LENS", count: 3}]);
+
+    vi.spyOn(overrides, "MOVESET_OVERRIDE", "get").mockReturnValue([Moves.WORRY_SEED, Moves.SOAK, Moves.TACKLE, Moves.TACKLE]);
+    vi.spyOn(overrides, "STARTING_HELD_ITEMS_OVERRIDE", "get").mockReturnValue([{name: "WIDE_LENS", count: 3}]);
+  });
+
+  it("create illusion at the start", async () => {
+    await game.startBattle([Species.ZOROARK, Species.AXEW]);
+
+    const zoroark = game.scene.getPlayerPokemon();
+    const zorua = game.scene.getEnemyPokemon();
+
+    expect(zoroark.illusion.active).equals(true);
+    expect(zorua.illusion.active).equals(true);
+    expect(zoroark.illusion.available).equals(false);
+
+  });
+
+  it("disappear after receiving damaging move and changing ability move", async () => {
+    await game.startBattle([Species.ZOROARK, Species.AXEW]);
+    game.doAttack(getMovePosition(game.scene, 0, Moves.WORRY_SEED));
+
+    await game.phaseInterceptor.to(TurnEndPhase);
+
+    const zoroark = game.scene.getPlayerPokemon();
+    const zorua = game.scene.getEnemyPokemon();
+
+    expect(zorua.illusion.active).equals(false);
+    expect(zoroark.illusion.active).equals(false);
+  });
+
+  it("disappear if the ability is suppressed", async () => {
+    vi.spyOn(overrides, "ABILITY_OVERRIDE", "get").mockReturnValue(Abilities.NEUTRALIZING_GAS);
+    await game.startBattle([Species.KOFFING]);
+
+    const zorua = game.scene.getEnemyPokemon();
+
+    expect(zorua.illusion.active).equals(false);
+  });
+
+  it("trick the enemy AI", async () => {
+    vi.spyOn(overrides, "OPP_MOVESET_OVERRIDE", "get").mockReturnValue([Moves.FLAMETHROWER, Moves.PSYCHIC, Moves.TACKLE, Moves.TACKLE]);
+    await game.startBattle([Species.ZOROARK, Species.AXEW]);
+
+    const enemy = game.scene.getEnemyPokemon();
+    const zoroark = game.scene.getPlayerPokemon();
+
+    const flameThwowerEffectiveness = zoroark.getAttackMoveEffectiveness(enemy, enemy.getMoveset()[0], false, true);
+    const psychicEffectiveness = zoroark.getAttackMoveEffectiveness(enemy, enemy.getMoveset()[1], false, true);
+
+    expect(psychicEffectiveness).above(flameThwowerEffectiveness);
```

Comment:
```suggestion
    const flameThrowerEffectiveness = zoroark.getAttackMoveEffectiveness(enemy, enemy.getMoveset()[0], false, true);
    const psychicEffectiveness = zoroark.getAttackMoveEffectiveness(enemy, enemy.getMoveset()[1], false, true);

    expect(psychicEffectiveness).above(flameThrowerEffectiveness);
```
NIT: Misspelled flamethrower

Reply:
my bad